frontend/public/components/namespace.jsx

```diff
@@ -831,43 +837,48 @@ export const ProjectsPage = connectToFlags(
 /** @type {React.SFC<{namespace: K8sResourceKind}>} */
 export const PullSecret = (props) => {
   const [isLoading, setIsLoading] = React.useState(true);
-  const [data, setData] = React.useState(undefined);
+  const [data, setData] = React.useState([]);
   const { t } = useTranslation();
+  const { namespace, canViewSecrets } = props;
 
   React.useEffect(() => {
-    k8sGet(SecretModel, null, props.namespace.metadata.name, {
-      queryParams: { fieldSelector: 'type=kubernetes.io/dockerconfigjson' },
-    })
-      .then((pullSecrets) => {
+    k8sGet(ServiceAccountModel, 'default', namespace.metadata.name, {})
+      .then((serviceAccount) => {
         setIsLoading(false);
-        setData(_.get(pullSecrets, 'items[0]'));
+        setData(serviceAccount.imagePullSecrets ?? []);
       })
       .catch((error) => {
         setIsLoading(false);
-        setData(undefined);
-        // A 404 just means that no pull secrets exist
-        if (error.status !== 404) {
-          throw error;
-        }
+        setData([]);
+        throw error;
```

Comment:
I know this was the behavior before, but I'm not sure we want to rethrow the error here. It would be better to show a message saying we couldn't get the service account. (Maybe an inline message under the property label?)

Reply:
Obviously, I was keeping what was there, but I also agree that showing a message is preferable vs a rethrow.  I'll look to see what other areas on the screen have in order to be consistent.

Reply:
Added an inline alert if there was an issue and removed the re-throw.